Panic for group by query over integer columns

<!-- Please describe your issue in English. -->

#### Can be reproduced ?
Yes

#### Steps:
DDL and DML are the same with #443
While keep inserting records until the total counts reach up to 1M

Query:
```
SELECT avg(LO_QUANTITY) FROM lineorder_flat GROUP BY LO_COMMITDATE;
```

#### Expected behavior:


#### Actual behavior:
Panic error:
```
query:select avg(LO_QUANTITY) from lineorder_flat group by LO_COMMITDATE
time of Exec.Compile : 66.124625ms
+++++++++
		:0 π(LO_COMMITDATE,LO_QUANTITY) -> γ([LO_COMMITDATE], [sumCount(LO_QUANTITY) -> avg(LO_QUANTITY)] -> transfer
		:0 π(LO_COMMITDATE,LO_QUANTITY) -> γ([LO_COMMITDATE], [sumCount(LO_QUANTITY) -> avg(LO_QUANTITY)] -> transfer
		:0 π(LO_COMMITDATE,LO_QUANTITY) -> γ([LO_COMMITDATE], [sumCount(LO_QUANTITY) -> avg(LO_QUANTITY)] -> transfer
		:0 π(LO_COMMITDATE,LO_QUANTITY) -> γ([LO_COMMITDATE], [sumCount(LO_QUANTITY) -> avg(LO_QUANTITY)] -> transfer
		:0 π(LO_COMMITDATE,LO_QUANTITY) -> γ([LO_COMMITDATE], [sumCount(LO_QUANTITY) -> avg(LO_QUANTITY)] -> transfer
		:0 π(LO_COMMITDATE,LO_QUANTITY) -> γ([LO_COMMITDATE], [sumCount(LO_QUANTITY) -> avg(LO_QUANTITY)] -> transfer
		:0 π(LO_COMMITDATE,LO_QUANTITY) -> γ([LO_COMMITDATE], [sumCount(LO_QUANTITY) -> avg(LO_QUANTITY)] -> transfer
		:0 π(LO_COMMITDATE,LO_QUANTITY) -> γ([LO_COMMITDATE], [sumCount(LO_QUANTITY) -> avg(LO_QUANTITY)] -> transfer
	:2 γ([LO_COMMITDATE], [sumCount(avg(LO_QUANTITY)) -> avg(LO_QUANTITY)] -> transfer
:1 γ([LO_COMMITDATE], [avg(avg(LO_QUANTITY)) -> avg(LO_QUANTITY)] -> π(avg(LO_QUANTITY)) -> sql output
+++++++++
panic: runtime error: index out of range [1043] with length 1024

goroutine 3290 [running]:
matrixone/pkg/container/vector.(*Vector).UnionOne(0xc00368f810, 0xc0017cf5e8, 0x413, 0x1)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/container/vector/vector.go:405 +0x46f4
matrixone/pkg/sql/colexec/group.(*Container).unitGroup(0xc003ad1b18, 0xc0036ee480, 0x400, {0x0, 0xc003ddd7d8, 0x0}, {0xc0001230c0, 0x54e1740, 0xc05c4fc101}, {0xc003f23200, ...}, ...)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/colexec/group/group.go:511 +0x7df
matrixone/pkg/sql/colexec/group.(*Container).batchGroup(0xc003694c80, {0xc0001230c0, 0x2, 0x2}, {0xc003f23200, 0x1, 0x1}, 0x203000)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/colexec/group/group.go:456 +0x16e
matrixone/pkg/sql/colexec/group.Call(0xc003ad4990, {0x535c220, 0xc003ad1b00})
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/colexec/group/group.go:159 +0x7f6
matrixone/pkg/vm.Run({0xc003afa320, 0x3, 0x2}, 0x2)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/vm/vm.go:35 +0x94
matrixone/pkg/vm/pipeline.(*Pipeline).Run(0xc003ddde78, {0xc05dab1600, 0x1, 0x1}, 0xc003ad4990)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/vm/pipeline/pipeline.go:62 +0x33b
matrixone/pkg/sql/compile.(*Scope).Run(0xc003f25100, {0x5917f98, 0xc0004204e8})
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/compile/scope.go:62 +0x6b9
matrixone/pkg/sql/compile.(*Scope).MergeRun.func1(0xc000226fb8)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/compile/scope.go:77 +0x33
created by matrixone/pkg/sql/compile.(*Scope).MergeRun
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/compile/scope.go:76 +0x412
panic: runtime error: index out of range [1200] with length 1024

goroutine 3289 [running]:
matrixone/pkg/container/vector.(*Vector).UnionOne(0xc00357aaf0, 0xc0035438f8, 0x4b0, 0x1)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/container/vector/vector.go:405 +0x46f4
matrixone/pkg/sql/colexec/group.(*Container).unitGroup(0xc003ad18d8, 0xc003ea01e0, 0x400, {0x0, 0xc003de17d8, 0x0}, {0xc00016db50, 0x54e1740, 0xc002783d01}, {0xc003f22940, ...}, ...)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/colexec/group/group.go:511 +0x7df
matrixone/pkg/sql/colexec/group.(*Container).batchGroup(0xc003566e80, {0xc00016db50, 0x2, 0x2}, {0xc003f22940, 0x1, 0x1}, 0x203000)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/colexec/group/group.go:456 +0x16e
matrixone/pkg/sql/colexec/group.Call(0xc003ad4900, {0x535c220, 0xc003ad18c0})
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/colexec/group/group.go:159 +0x7f6
matrixone/pkg/vm.Run({0xc003afa2d0, 0x3, 0x2}, 0x2)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/vm/vm.go:35 +0x94
matrixone/pkg/vm/pipeline.(*Pipeline).Run(0xc003de1e78, {0xc05dab1580, 0x1, 0x1}, 0xc003ad4900)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/vm/pipeline/pipeline.go:62 +0x33b
matrixone/pkg/sql/compile.(*Scope).Run(0xc003f25080, {0x5917f98, 0xc0004204e8})
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/compile/scope.go:62 +0x6b9
matrixone/pkg/sql/compile.(*Scope).MergeRun.func1(0x0)
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/compile/scope.go:77 +0x33
created by matrixone/pkg/sql/compile.(*Scope).MergeRun
	/Users/yingfeng/codebase/matrixorigin/yingfeng/matrixone/pkg/sql/compile/scope.go:76 +0x412
```

#### Environment:
- Version or commit-id (e.g. v0.1.0 or d4989d4):
- Hardware parameters:
- OS type: MAC
- Others:

#### Configuration file:
AOE storage is manually configured in pkg/vm/engine/aoe/dist/aoe/storage.go : 
```
func NewStorage(dir string) (*Storage, error) {
	opts := &store.Options{}
    cfg := &md.Configuration{
        Dir:              dir,
        BlockMaxRows:     10000,
        SegmentMaxBlocks: 8,
    }
    cacheCfg := &store.CacheCfg{
        IndexCapacity:  1024 * 1024 * 1024,
        InsertCapacity: 1024 * 1024 * 1024,
        DataCapacity:   4 * 1024 * 1024 * 1024,
    }
    opts.Meta.Conf = cfg
    opts.CacheCfg = cacheCfg
	return NewStorageWithOptions(dir, opts)
}
```


#### Additional context:
- Error message from client:
- Server log:
- Other information:

--- a/pkg/container/vector/vector.go
+++ b/pkg/container/vector/vector.go
@@ -159,24 +159,86 @@ func (v *Vector) Length() int {
 
 func (v *Vector) Window(start, end int) *Vector {
 	switch v.Typ.Oid {
+	case types.T_int8:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]int8)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_int16:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]int16)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_int32:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]int32)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_int64:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]int64)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_uint8:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]uint8)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_uint16:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]uint16)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_uint32:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]uint32)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_uint64:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]uint64)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_float32:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]float32)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_float64:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]float64)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_sel:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([]int64)[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
+	case types.T_tuple:
+		return &Vector{
+			Typ: v.Typ,
+			Col: v.Col.([][]interface{})[start:end],
+			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
+		}
 	case types.T_char, types.T_varchar, types.T_json:
 		return &Vector{
 			Typ: v.Typ,
 			Col: v.Col.(*types.Bytes).Window(start, end),
 			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
 		}
-	default:
-		col := v.Col
-		ptr := (*(*emptyInterface)(unsafe.Pointer(&col))).word
-		data := *(*uintptr)(unsafe.Pointer(uintptr(ptr)))
-		*(*uintptr)(unsafe.Pointer(uintptr(ptr))) = data + uintptr(v.Typ.Size)*uintptr(start)
-		*(*int)(unsafe.Pointer(uintptr(ptr) + uintptr(strconv.IntSize>>3))) = end - start
-		return &Vector{
-			Typ: v.Typ,
-			Col: col,
-			Nsp: v.Nsp.Range(uint64(start), uint64(end)),
-		}
 	}
+	return nil
 }
 
 func (v *Vector) Append(arg interface{}) error {

